### DISCUSSION TITLE
Failed to Push Branch to pmd github project

### DISCUSSION THREAD

**Discussion by nathanspectacular** [pmd/pmd#3476] [Q&A] [answered]
I wrote a new rule's code and test.  I would like to push my branch and create a pull request.  I am using the following command to push the branch.

`git push github AvoidAssertRule`

I am getting the following error:

```
ERROR: Permission to pmd/pmd.git denied to nathanspectacular.
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.
```

Do I need to gain permission to push to the pmd project?  Do I need to fork pmd, push to my fork and then create a pull request from my fork to pmd?

**Comment by adangel** [comment, marked as answer]
> Do I need to fork pmd, push to my fork and then create a pull request from my fork to pmd?

Exactly this, yes - that's how github works.

See also https://github.com/pmd/pmd/wiki/Newcomers%27-Guide
